src/category_theory/limits/shapes/finite_products.lean

```diff
@@ -39,6 +39,13 @@ by { haveI := @has_finite_products.out C _ _ J (classical.dec_eq _), apply_insta
 instance has_finite_products_of_has_finite_limits [has_finite_limits C] : has_finite_products C :=
 ⟨λ J 𝒥₁ 𝒥₂, by { resetI, apply_instance }⟩
 
+instance has_fin_products [has_finite_products C] (n : ℕ) :
+  has_limits_of_shape (discrete (fin n)) C :=
+has_limits_of_shape_of_equivalence (discrete.equivalence (show ulift.{v} (fin n) ≃ fin n, by tidy))
+
+/-- We can now write this for powers. -/
+noncomputable example [has_finite_products C] (X : C) : C := ∏ (λ (i : fin 5), X)
+
```

Comment:
We can now do this as a plus.
Some further work needs to be done before this could be used for biproducts though.
Since the `bicones` library seem to lack some constructions (in particular composing with equivalences).

Reply:
It's really cool that this is now possible!